pkg/portal/middleware/aad.go

```diff
@@ -141,7 +142,18 @@ func (a *aad) AAD(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		session, err := a.store.Get(r, SessionName)
 		if err != nil {
-			a.internalServerError(w, err)
+			cookieError := err.(securecookie.Error)
```

Comment:
should we handle the type assertion? otherwise the portal could panic here?
something like this perhaps: 
```suggestion
                        cookieError, ok := err.(securecookie.Error)
                        if ok && cookieError != nil && cookieError.IsDecode() {
```

Reply:
wouldn't 
```suggestion
			cookieError, ok  := err.(securecookie.Error)
			if !ok {
			    a.internalServerError(w, err)
			    return
			}
```
be more explicit ?